File: Napakalaki/lib/bad_consequence.rb
# encoding: utf-8

module NapakalakiGame

  # class BadConsequence: representa el mal rollo de un monstruo
  class BadConsequence

    def initialize(t, l)
      @text = t
      @levels = l
    end
    
    def getLevels
      @levels
    end
    
    def to_s
      text = "    * Texto: #{@text}"
      text << "\n    * Niveles perdidos: #{@levels}"
    end

  end # BadConsequence
  
end # NapakalakiGame

File: Napakalaki/src/NapakalakiGame/BadConsequence.java
/**
 * BadConsequence.java
 */

package NapakalakiGame;

import java.util.ArrayList;

/**
 * Representa un mal rollo genérico de un monstruo.
 */
public abstract class BadConsequence {

    private String text;
    private int levels;
    
    public BadConsequence(String t, int l) {
        text = t;
        levels = l;
    }

    public int getLevels() {
        return levels;
    }
    
    public String getText() {
        return text;
    }
    
    public abstract boolean isEmpty();
    
    protected abstract void substractVisibleTreasure(Treasure t);
    
    protected abstract void substractHiddenTreasure(Treasure t);
    
    protected abstract BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, 
                                                            ArrayList<Treasure> h);
    
    @Override
    public String toString() {
        return "Niveles perdidos: " + Integer.toString(levels);
    }
    
}
